pytest/lib/mocknet.py

```diff
@@ -524,146 +579,169 @@ def create_genesis_file(validator_node_names,
                             record['Account']['account']['locked'] = str(0)
 
     else:
-        genesis_config['total_supply'] = str(TOTAL_SUPPLY)
-        master_balance = (TOTAL_SUPPLY - add_balance)
-        assert master_balance > 0
         genesis_config['records'] = []
-        genesis_config['records'].append({
-            "Account": {
-                "account_id": MASTER_ACCOUNT,
-                "account": {
-                    "amount": str(master_balance),
-                    "locked": "0",
-                    "code_hash": "11111111111111111111111111111111",
-                    "storage_usage": 0,
-                    "version": "V1"
-                }
-            }
-        })
-        genesis_config['records'].append({
-            "AccessKey": {
-                "account_id": MASTER_ACCOUNT,
-                "public_key": PUBLIC_KEY,
-                "access_key": {
-                    "nonce": 0,
-                    "permission": "FullAccess"
-                }
-            }
-        })
-        genesis_config['records'].append({
-            "Account": {
-                "account_id": TREASURY_ACCOUNT,
-                "account": {
-                    "amount": str(TREASURY_BALANCE),
-                    "locked": "0",
-                    "code_hash": "11111111111111111111111111111111",
-                    "storage_usage": 0,
-                    "version": "V1"
+
+    master_balance = 10**7
+    assert master_balance > 0
+    accounts_and_balances = [(TREASURY_ACCOUNT, TREASURY_BALANCE),
+                             (MASTER_ACCOUNT, master_balance),
+                             (SKYWARD_ACCOUNT, SKYWARD_CONTRACT_BALANCE),
+                             (TOKEN1_ACCOUNT, TOKEN1_BALANCE),
+                             (TOKEN2_ACCOUNT, TOKEN2_BALANCE),
+                             (TOKEN2_OWNER_ACCOUNT, TOKEN2_OWNER_BALANCE),
+                             (ACCOUNT1_ACCOUNT, ACCOUNT1_BALANCE)]
+    accounts = {}
+    for a in accounts_and_balances:
+        accounts[a[0]] = a[1]
+    seen_accounts = set()
+    for record in genesis_config['records']:
+        if 'Account' in record and 'account_id' in record['Account'] and record[
+                'Account']['account_id'] in accounts:
+            seen_accounts.add(record['Account']['account_id'])
+            if 'account' in record['Account'] and 'amount' in record['Account'][
+                    'account']:
```

Comment:
So if `amount` is specified we want to override it? But if it’s not specified we want to leave it unspecified? Shouldn’t it be `not in`?

And what if `account` is not set? Should we set it instead of skipping the `account` for given `Account`?